iotdb-client/client-py/iotdb/utils/IoTDBRpcDataSet.py

```diff
@@ -142,30 +137,141 @@ def close(self):
 
             self.__is_closed = True
             self.__client = None
-            if isinstance(self.__query_data_set.time, memoryview):
-                self.__query_data_set.time.release()
-            for value in self.__query_data_set.valueList:
-                if isinstance(value, memoryview):
-                    value.release()
-            for bitmap in self.__query_data_set.bitmapList:
-                if isinstance(bitmap, memoryview):
-                    bitmap.release()
 
     def next(self):
-        if self.has_cached_result():
-            self.construct_one_row()
+        if not self.has_cached_data_frame:
+            self.construct_one_data_frame()
+        if self.has_cached_data_frame:
             return True
         if self.__empty_resultSet:
             return False
         if self.fetch_results():
-            self.construct_one_row()
+            self.construct_one_data_frame()
             return True
         return False
 
-    def has_cached_result(self):
-        return (self.__query_data_set is not None) and (
-            len(self.__query_data_set.time) != 0
```

Comment:
Should we keep len(self.__query_data_set.time) != 0? Maybe the resultSet is already an emptySet.

Reply:
Fixed